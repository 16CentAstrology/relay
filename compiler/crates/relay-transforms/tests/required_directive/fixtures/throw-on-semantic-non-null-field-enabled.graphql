# relay:feature-flag disallow_required_action_throw_on_semantically_nullable_fields

query TestQuery {
  opera {
    # This field has @semanticNonNull, so THROW should work when feature flag is enabled
    composer @required(action: THROW) {
      name
    }
  }
}
