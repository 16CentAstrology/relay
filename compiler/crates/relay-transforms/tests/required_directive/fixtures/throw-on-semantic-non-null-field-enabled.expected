==================================== INPUT ====================================
# relay:feature-flag disallow_required_action_throw_on_semantically_nullable_fields

query TestQuery {
  opera {
    # This field has @semanticNonNull, so THROW should work when feature flag is enabled
    composer @required(action: THROW) {
      name
    }
  }
}
==================================== OUTPUT ===================================
query TestQuery {
  opera {
    composer @required(action: THROW) @__RequiredMetadataDirective
    # RequiredMetadataDirective {
    #     action: Throw,
    #     path: "opera.composer",
    # }
     {
      name
    }
  }
}
